tests/test_utils.py

```diff
@@ -136,3 +138,49 @@ def test_encoding_oversized_recursive(self):
                          {"Test": ["a" * 1024, "b", "c"],
                           "Self": "[RECURSIVE]",
                           "Other": {"a": 300}})
+
+    def test_encoding_time(self):
+        """
+        Test that encoding a large object is sufficiently speedy
+        """
+        setup = """\
+import json
+from tests.large_object import large_object_file_path
+from bugsnag.utils import SanitizingJSONEncoder
+encoder = SanitizingJSONEncoder(keyword_filters=[])
+with open(large_object_file_path()) as json_data:
+    data = json.load(json_data)
+        """
+        stmt = """\
+encoder.encode(data)
+        """
+        time = timeit.timeit(stmt=stmt, setup=setup, number=1000)
+        maximum_time = 6
+        if sys.version_info[0:2] <= (2, 6):
+            # json encoding is very slow on python 2.6 so we need to increase
+            # the allowable time when running on it
+            maximum_time = 18
+        self.assertTrue(time < maximum_time,
```

Comment:
I don't like this as a pass/fail metric for tests, as it's too dependent on the performance of the CI tools.

Reply:
Its dependent on the tools, but also a small canary if something goes terribly wrong under the lax requirements of this test. I'd prefer that we have something for now and look into something more thorough in the future, like a formal perf suite.